### PR TITLE
README: Update with CMake and windows example

### PR DESCRIPTION
**Pull request by adithyab94**
Added example to build the project using CMake since lot of people have been enquiring about this

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -145,6 +145,9 @@ git clone https://github.com/ggerganov/llama.cpp
 cd llama.cpp
 make
 
+#For Windows and CMake, use the following command instead:
+cmake -S . -B build/ -G "MinGW Makefiles"
```
Comment (howard0su) [inline README.md:149]: this only works for mingw. but installing a vs community edition is much simpler for average users.

Comment (KASR) [reply README.md:149]: Yes indeed, you could add the following lines for the people using vs and cmake:

```
cd <path_to_llama_folder>
mkdir build
cd build
cmake ..
cmake --build . --config Release
```

Comment (adithyab94) [reply README.md]: Thanks, I have updated it with your suggestion

File: README.md [hunk 2]
```diff
@@ -145,6 +145,13 @@ git clone https://github.com/ggerganov/llama.cpp
 cd llama.cpp
 make
 
+#For Windows and CMake, use the following command instead:
```
Comment (howard0su) [inline README.md:148]: the instruction looks good. can we also mentioned that the user needs to install visual studio?

**Review by ggerganov — APPROVED** [review]

**Comment by ggerganov** [comment]
@howard0su For now we will keep the Makefile since it is the most trivial way to build the project and a lot of developers appreciate the simplicity

**Comment by td-us** [comment]
Hi, I tried to follow your updated guidelines to run the model. Unfortunately, I encounter the error "Error: could not load cache" when using command "cmake --build . --config Release". Any idea why please? I am using Windows 10, installed git, cmake, visual studio and anaconda, am using anaconda prompt. cmake .. gives this output :
image

**Pull request merged**